main.js

```diff
@@ -0,0 +1,147 @@
+const gameBoard = (() => {
+  //Fill the board 3x3 values
+  const board = fillBoard();
+
+  const render = () => {
+    let boardHTML = "";
+    board.forEach((square, position) => {
+      boardHTML += `<div class="squares" id="${position}"> ${square} </div>`;
+    });
+
+    document.querySelector(".gameboard").innerHTML = boardHTML;
+    const squares = document.querySelectorAll(".squares");
+    squares.forEach((square) => {
+      square.addEventListener("click", Game.handleClick);
+    });
+  };
+
+  const update = (position, value) => {
+    board[position] = value;
+    render();
+  };
+
+  const getGameboard = () => board;
+  return {
+    render,
+    update,
+    getGameboard,
+  };
+})();
+
+const Game = (() => {
+  let players = [];
+  let currentPlayerIndex;
+  let gameOver;
+
+  const start = () => {
+    players = [
+      createPlayer(document.querySelector(".player1").value, "X"),
+      createPlayer(document.querySelector(".player2").value, "O"),
+    ];
+    currentPlayerIndex = 0;
+    gameOver = false;
+    gameBoard.render();
+
+    const squares = document.querySelectorAll(".squares");
+    squares.forEach((square) => {
+      square.addEventListener("click", Game.handleClick);
+    });
+  };
+
+  const handleClick = (event) => {
+    if (gameOver) {
+      return;
+    }
+    let index = Number(event.target.id);
+
+    if (gameBoard.getGameboard()[index] !== "") {
+      currentPlayerIndex = currentPlayerIndex === 0 ? 1 : 0;
+      return;
+    }
+
+    const curPlayer = players[currentPlayerIndex].name;
```

Comment:
rename `curPlayer` to `currentPlayer` and `curMark` to `currentMark`, as its more readable.

Reply:
Fixed